### DISCUSSION TITLE
Add CLI cache settings to config file

### DISCUSSION THREAD

**Discussion by paulshryock** [prettier/prettier#13581] [Ideas]
**Edit 10/16/2025**: I have created an issue for this idea: https://github.com/prettier/prettier/issues/18089

---

The CLI cache settings are really nice.

I'd love to be able to configure these from a Prettier configuration file instead of from the CLI.

- [`--cache`](https://prettier.io/docs/en/next/cli.html#--cache)
- [`--cache-location`](https://prettier.io/docs/en/next/cli.html#--cache-location)
- [`--cache-strategy`](https://prettier.io/docs/en/next/cli.html#--cache-strategy)

Like this:

```javascript
// prettier.config.cjs

module.exports = {
  cache: true,
  cacheLocation: '.cache/prettier/.prettier-cache',
  cacheStrategy: CI ? 'content' : 'metadata',
}
```

This way the caching strategy can be set dynamically for CI vs local, and my `package.json` can be a lot cleaner:

From this:
```json
{
  "scripts": {
    "format": "prettier --cache --cache-location=.cache/prettier/.prettier-cache",
    "format:ci": "npm run format -- --cache-strategy=content",
    "format:local": "npm run format -- --cache-strategy=metadata",
  }
}
```

To this:
```json
{
  "scripts": {
    "format": "prettier"
  }
}
```

I wanted to get some thoughts and opinions from the maintainers and community first. I would be happy to write up an issue, and possibly a PR, if there is a positive consensus about this.

**Comment by eeriemyxi** [comment]
I have no idea why this isn't a thing to this day. To me it is such an obvious feature that Prettier should have.

**Reply by paulshryock** [reply]
Been a few years, and never got any momentum. 7 likes and 1 comment. I could create an issue to request this more formally, but again not sure if there's enough community interest. 🤷

**Reply by sirimykland** [reply]
I would really like to have this feature

**Reply by paulshryock** [reply]
I have created an issue for this idea: https://github.com/prettier/prettier/issues/18089
